### PR TITLE
[Capture Project] Easier access to custom primitives

### PR DESCRIPTION
**Pull request by albi3ro**
When prototyping and exploring the capture project, I was finding it extremely annoying to track down all the different types of primitives that are currently scattered throughout the code base, and remember where I needed to import them all from.

To make developement easier, I created a new `primitives` submodule that we can just import them all from.  This submodule is not imported by default with pennylane, so we do not need to worry about import paths.

But now we can do things like:
```
from pennylane.capture.primitives import *
from pennylane.capture.primitives import for_loop_prim
```

which will be substantially easier than the current route:

```
from pennylane.compiler.qjit_api import _get_for_loop_qfunc_prim
for_loop_prim = _get_for_loop_qfunc_prim
```

Since `capture.primitives` was already a bunch of operator and measurement code, I moved that code to `capture_operators` and `capture_measurements` respectively.l

### CODE REVIEW

File: doc/releases/changelog-dev.md [hunk 1]
```diff
@@ -158,6 +158,9 @@
   [(#6041)](https://github.com/PennyLaneAI/pennylane/pull/6041)
   [(#6064)](https://github.com/PennyLaneAI/pennylane/pull/6064)
 
+* Capture project objects with jax dependencies can now be accessed via the `pennylane.capture.primitives`
+  submodule.  This module is not automatically imported with pennylane.
```
Comment (albi3ro) [inline doc/releases/changelog-dev.md:162]:
```suggestion
  submodule.  This module is not automatically imported with pennylane.
  [(#6129)](https://github.com/PennyLaneAI/pennylane/pull/6129)
```

File: pennylane/capture/__init__.py [hunk 2]
```diff
@@ -35,6 +35,25 @@
     ~create_measurement_mcm_primitive
     ~qnode_call
 
+
+The ``primitives`` submodule is offers easy access to objects with jax dependencies.
+It is not available with ``import pennylane``, but the contents can be access via manual
```
Comment (dwierichs) [inline pennylane/capture/__init__.py:40]:
```suggestion
The ``primitives`` submodule offers easy access to objects with jax dependencies such as
primitives and abstract types.
It is not available with ``import pennylane``, but the contents can be accessed via manual
```

File: pennylane/capture/primitives.py [hunk 3]
```diff
@@ -12,309 +12,38 @@
...
-        return type.__call__(measurement_type, wires=wires, **kwargs)
-
-    abstract_type = _get_abstract_measurement()
-
-    @primitive.def_abstract_eval
-    def _(*args, has_eigvals=False, **_):
-        abstract_eval = measurement_type._abstract_eval  # pylint: disable=protected-access
-        n_wires = len(args) - 1 if has_eigvals else len(args)
-        return abstract_type(abstract_eval, n_wires=n_wires, has_eigvals=has_eigvals)
+It has a jax dependency and should be be located in a standard import path.
```
Comment (dwierichs) [inline pennylane/capture/primitives.py:18]:
```suggestion
It has a jax dependency and should be located in a standard import path.
```

**Review by dwierichs — COMMENTED** [review]
Nice, this looks super helpful!

- We will need to adapt Catalyst's `from_plxpr`, in particular [this line](https://github.com/PennyLaneAI/catalyst/blob/b95d8622b133071d02a08783ccba4f736d4d43f9/frontend/catalyst/from_plxpr.py#L22), I think?
- Could it be confusing that `AbstractOperator` and `AbstractMeasurement` are still accessible from `pennylane.capture` whereas the primitives are not?
- Otherwise only found some tiny text things to comment on.

**Comment by albi3ro** [comment]
> Nice, this looks super helpful!
> 
> * We will need to adapt Catalyst's `from_plxpr`, in particular [this line](https://github.com/PennyLaneAI/catalyst/blob/b95d8622b133071d02a08783ccba4f736d4d43f9/frontend/catalyst/from_plxpr.py#L22), I think?
> * Could it be confusing that `AbstractOperator` and `AbstractMeasurement` are still accessible from `pennylane.capture` whereas the primitives are not?
> * Otherwise only found some tiny text things to comment on.

Readded `qml.capture.qnode_prim` so we don't have to worry about updating catalyst for the moment. Thanks for catching that.

I left `AbstractOperator` accessible as `qml.capture.AbstractOperator`, because `MeasurementProcess` was using there and I didn't feel like changing that code.

**Review by dwierichs — APPROVED** [review]
Thanks @albi3ro for addressing all concerns :) 
LGTM! :rocket:

File: doc/releases/changelog-dev.md [hunk 4]
```diff
@@ -6,6 +6,10 @@
 
 <h3>Improvements 🛠</h3>
 
+* Some custom primitives can for the capture project can now be imported via
+  `from pennylane.capture.primitives import *`.
```
Comment (PietropaoloFrisoni) [inline doc/releases/changelog-dev.md:10]:
```suggestion
* Some custom primitives for the capture project can now be imported via
  `from pennylane.capture.primitives import *`.
```

**Review by PietropaoloFrisoni — APPROVED** [review]
Thanks @albi3ro for making this easier! I only noticed a typo in the changelog

**Pull request merged**